CMSgov/onemac
# Oy2 81 user submits new waiver

Changes:
1. Created waivers folder to follow pattern of amendments
2. Created Waivers.js and NewWaiver.js follow to pattern of Amendments.js and NewAmendments.js
3. Updated Waivers and NewWaivers to contain  the following elements: State, Action Type, Waiver Number, Waiver Authority, Summary as described in ticket.
4. Added Waivers and NewWaiver to the Route.js
5. Created libs/waiverLib.js to contain selectable options.
6. Updated index.scss so that Waivers are styled like Amendments.
7. Added “waivers” to the amplify API
8. Updated Home.js to be able to show Waivers list and select “Add New Waiver” - had to break the dependency on Amendments as the single display type. I recognize that this is something I did to test/ully realize how to best separate Waivers from Amendments and was NOT part of the original ticket.
9. Created createWaiver.js, listWaiver.js, and getWaiver.js to follow pattern of amendment backend solution and modified to reflect waiver differences.
10. Updated serverless configuration to add waiver handling
11. Modified list.js to exclude waivers by using a not-waivertype filter (so base type is amendment for now) (this was done to not break existing codebase)

Test:
1. Log in to form.  Verify home page has “Submit New Waiver”
2. Click “Submit New Waiver” and verify that the form page loads, the fields meet the Acceptance criteria, and the Submit button is disabled.
3. Start filling out a new Waiver submission. Verify that the submit button is still disabled until all required fields (State, Action Type, Waiver Number, Waiver Authority) are entered.
4. Submit the Waiver.
5. Use the DynamoDB interface in the AWS console though CloudTamer to verify that the waiver data is stored correctly.
6. Verify that SPA amendment submissions still work as expected.
